drivers/i2c/i2c_gecko.c

```diff
@@ -37,6 +37,7 @@ struct i2c_gecko_config {
 
 struct i2c_gecko_data {
 	struct k_sem device_sync_sem;
+	struct k_sem bus_mutex;
```

Comment:
bus_lock might be more appropriate as a name here, as its not a k_mutex

Reply:
bus_lock is more explicite, done